File: CK.Ng.UserProfile/CK.Cris.SimpleBatch/ICommandSimpleBatch.cs
using CK.Core;

namespace CK.Cris;

// [Abstract] -> should not be sent, must be specified
public interface ICommandSimpleBatch : ICommandPart, ICommand<ISimpleBatchCommandResultPart>
{
    public IList<(IAbstractCommand Command, string? Description)> Commands { get; }
}

[CKTypeDefiner]
// [Virtual]
public interface ISimpleBatchCommandResultPart : IStandardResultPart
{
    public IList<IPocoCommandExecutedCommandResult> Results { get; }

    //public IList<IPocoCommandExecutedPart> Results { get; }
}

// Will disappear once we have [Abstract], [Virtual], [Primary], [Secondary]
public interface IPocoCommandExecutedCommandResult : IPocoCommandExecutedPart { }

File: Tests/CK.Ng.UserProfile.Workspace.Tests/TSInlineTests/CK_Ng_UserProfile_Workspace/ck-gen/CK/Cris/ICommandSimpleBatch.ts
import { IAbstractCommand, ICommand } from './Model';
import { ICommandPart } from './ICommandPart';
import { ISimpleBatchCommandResultPart } from './ISimpleBatchCommandResultPart';

export interface ICommandSimpleBatch extends ICommandPart, ICommand<ISimpleBatchCommandResultPart|undefined> {
readonly commands: Array<{command: IAbstractCommand, description?: string}>;
readonly _brand: ICommandPart["_brand"] & ICommand<ISimpleBatchCommandResultPart|undefined>["_brand"] & {"43":any};
}
